tools/building.py

```diff
@@ -1006,19 +1004,43 @@ def add_to_path(dirname):
     # Tell closure never to inject the 'use strict' directive.
     args += ['--emit_use_strict=false']
 
+    # Closure compiler is unable to deal with path names that are not 7-bit ASCII:
+    # https://github.com/google/closure-compiler/issues/3784
+    outfile = configuration.get_temp_files().get('.cc.js').name  # Safe 7-bit filename
```

Comment:
I think its best to minimize the number of calls to `get_temp_files()` and share the return value instead.  

```
tmpfiles = configuration.get_temp_files()
outfile = tempfiles.get('.cc.js').name
```

As well as doing less work this also prints less stuff to the log (each TempFiles instance prints `not cleaning up temp files since in debug-save mode, see them ..` to stdout in save temps mode).   We should really fix that but that is completely unrelated to this change.

You can also then do `tempfiles.tmpdir` below to avoid changing `shared.py` at all.